app/src/main/java/com/picone/go4lunch/presentation/ui/AuthenticationFragment.java

```diff
@@ -1,41 +1,141 @@
 
     @Nullable
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
         mBinding = FragmentAuthenticationBinding.inflate(inflater, container, false);
-        mBinding.loginWithFacebook.setOnClickListener(v -> goToMaps());
-        mBinding.loginWithGoogle.setOnClickListener(v -> goToMaps());
+        initView();
+        setNavVisibility(false);
         return mBinding.getRoot();
     }
 
-    private void goToMaps() {
-        NavHostFragment.findNavController(this).navigate(R.id.action_authenticationFragment_to_mapsFragment);
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        initAuthenticationAborted();
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        mCallbackManager.onActivityResult(requestCode, resultCode, data);
+        super.onActivityResult(requestCode, resultCode, data);
+        if (requestCode == RC_SIGN_IN) {
+            Task<GoogleSignInAccount> task = GoogleSignIn.getSignedInAccountFromIntent(data);
+            try {
+                GoogleSignInAccount account = task.getResult(ApiException.class);
+                assert account != null;
+                firebaseAuthWithGoogle(account.getIdToken());
+            } catch (ApiException e) {
+                Toast.makeText(getContext(), R.string.google_auth_failed, Toast.LENGTH_SHORT).show();
+            }
+        }
+    }
+
+    private void initView() {
+        mBinding.loginWithFacebook.setOnClickListener(v -> signInWithFacebook());
+        mBinding.loginWithGoogle.setOnClickListener(v -> signInWithGoogle());
+    }
+
+    private void initAuthenticationAborted() {
+        requireActivity().getOnBackPressedDispatcher().addCallback(getViewLifecycleOwner(),
+                new OnBackPressedCallback(true) {
+                    @Override
+                    public void handleOnBackPressed() {
+                        mLoginViewModel.refuseAuthentication();
+                    }
+                });
+    }
+    //--------------------Authentication with google----------------------------
+
+    private void signInWithGoogle() {
+        Intent signInIntent = mGoogleSignInClient.getSignInIntent();
+        startActivityForResult(signInIntent, RC_SIGN_IN);
+    }
+
+    private void firebaseAuthWithGoogle(String idToken) {
+        AuthCredential credential = GoogleAuthProvider.getCredential(idToken, null);
+        mAuth.signInWithCredential(credential)
+                .addOnCompleteListener(requireActivity(), task -> {
+                    if (task.isSuccessful()) {
+                        FirebaseUser firebaseUser = mAuth.getCurrentUser();
+                        assert firebaseUser != null;
+                        mLoginViewModel.authenticate(true);
```

Comment:
Remove these two following lines.
You've already subscribed to the livedata in MainActivity